### PR TITLE
View/apikeys

### PR DESCRIPTION
**Pull request by MoritzHahn1337**

### CODE REVIEW

**Comment by MoritzHahn1337** [comment]
The most recent commit also fixes #139

**Comment by KerstenBreuer** [comment]
@lkuchenb, could you have a look at this, I think I need this PR to be merged to carry on.

**Review by lkuchenb — CHANGES_REQUESTED** [review]

**Review by lkuchenb — APPROVED** [review]

File: datameta/api/apikeys.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __json__(self, request: Request) -> dict:
                 "email": self.email,
                 "token": self.token,
                 "label": self.label,
-                "expiresAt": self.expires_at.isoformat() if self.expires_at else None
```
Comment (KerstenBreuer) [inline datameta/api/apikeys.py:54]: We cannot return the expires_at directly, as it is a datetime object, however, openapi expects string as a response. I could also suggest to format the datetime object in isoformat to stay consistent.

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]

**Comment by KerstenBreuer** [comment]
@MoritzHahn1337, if you agree with the above, I could also make the changes. You would just need to ensure that the frontend send isoformated expiration dates. Would that be OK?

**Comment by MoritzHahn1337** [comment]
@KerstenBreuer Okay, I will make sure the frontend sends in isoformat. Would you pack changing the backend into #146 or would you stay in this PR?

**Comment by KerstenBreuer** [comment]
That's a good idea, I will include that, just let me know once the required changes in the front end are in place

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>

________________________________

**Comment by MoritzHahn1337** [comment]
The most recent commit should take care of that.

**Comment by KerstenBreuer** [comment]
Cool, thanks a lot @MoritzHahn1337. I will take it from there.

**Review by KerstenBreuer — APPROVED** [review]

**Comment by KerstenBreuer** [comment]
@lkuchenb, @MoritzHahn1337 can you have a brief look at it again?

This implements the ApiKey expiration checking on each endpoint.
There is a new config parameter `datameta.apikeys.max_expiration_period` for defining the max allowed period a token is allowed to valid (added to all .ini files with 30 days as default). A user may choose to further restrict the validity of a token by specifying a expiration date that is earlier than `datameta.apikeys.max_expiration_period`.

**Comment by KerstenBreuer** [comment]
Will fix #146

File: datameta/security.py [hunk 2]
```diff
@@ -21,13 +21,29 @@
...
 
 from .models import User, ApiKey
 
 import bcrypt
 import logging
 log = logging.getLogger(__name__)
 
+def check_expiration(
+    expiration_datetime:Optional[datetime], 
+    datetime_to_check:Optional[datetime]=None
```
Comment (lkuchenb) [inline datameta/security.py:35]: What's the use case for not using `now()`?

Comment (KerstenBreuer) [reply datameta/security.py:35]: Can be remove

**Pull request merged**